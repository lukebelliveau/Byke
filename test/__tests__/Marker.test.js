import React from 'react';
import renderer from 'react-test-renderer';

import Marker from '../../src/StationMarker';

describe('Marker', () => {
  const coordinate = { latitude: 50, longitude: 50 };
  it('should be green', () => {
    const marker = renderer
      .create(
        <Marker
          coordinate={coordinate}
          stationName="station"
          availableBikes={8}
        />
      )
      .toJSON();

    expect(marker).toMatchSnapshot();
  });

  it('should be red', () => {
    const marker = renderer
      .create(
        <Marker
          coordinate={coordinate}
          stationName="station"
          availableBikes={4}
        />
      )
      .toJSON();

    expect(marker).toMatchSnapshot();
  });
});
