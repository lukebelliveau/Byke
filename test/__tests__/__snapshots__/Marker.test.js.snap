// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker should be green 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "latitude": 50,
      "longitude": 50,
    }
  }
  description="8 bikes available"
  image={undefined}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      undefined,
    ]
  }
  title="station"
>
  <View
    style={
      Object {
        "backgroundColor": "green",
        "borderRadius": 10,
        "height": 20,
        "width": 20,
      }
    }
  />
</AIRMapMarker>
`;

exports[`Marker should be red 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "latitude": 50,
      "longitude": 50,
    }
  }
  description="4 bikes available"
  image={undefined}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      undefined,
    ]
  }
  title="station"
>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 10,
        "height": 20,
        "width": 20,
      }
    }
  />
</AIRMapMarker>
`;
